test/ExtendedXmlSerializer.Tests/ReportedIssues/Issue141Tests.cs

```diff
@@ -0,0 +1,82 @@
+﻿using ExtendedXmlSerializer;
+using ExtendedXmlSerializer.Configuration;
+using ExtendedXmlSerializer.ExtensionModel;
+using ExtendedXmlSerializer.ExtensionModel.Content;
+using ExtendedXmlSerializer.ExtensionModel.Xml;
+using FluentAssertions;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace ExtendedXmlSerializer.Tests.ReportedIssues
+{
+    public class Issue141Tests
+    {
+        [Fact]
+        public void ShouldPreserveNullStringValueIfDefaultIsNotNull()
+        {
+            var config = new ConfigurationContainer();
+            config.Emit(EmitBehaviors.Assigned); //no matter what we put here
+            var serializer = config.Create();
+
+            string xml = serializer.Serialize(new ClassWithDefautString() { Name = null, SubNode = null });
+            var deserialized = serializer.Deserialize<ClassWithDefautString>(xml);
+
+            deserialized.Name.Should().BeNull();    //fail
+        }
+
+        [Fact]
+        public void ShouldPreserveNullObjectValueIfDefaultIsNotNull()
+        {
+            var config = new ConfigurationContainer();
+            config.Emit(EmitBehaviors.Assigned); //no matter what we put here
+            var serializer = config.Create();
+
+            string xml = serializer.Serialize(new ClassWithDefautString() { Name = null, SubNode = null });
+            var deserialized = serializer.Deserialize<ClassWithDefautString>(xml);
+
+            deserialized.SubNode.Should().BeNull();    //fail
+        }
+
+        [Fact]
+        public void ShouldPreserveNullAttributeValueIfDefaultIsNotNull()
+        {
+            var config = new ConfigurationContainer();
+            config.Emit(EmitBehaviors.Assigned); //no matter what we put here
+            config.ConfigureType<ClassWithDefautString>().Member(x => x.Attribute).Attribute();
+            var serializer = config.Create();
+
+            string xml = serializer.Serialize(new ClassWithDefautString() { Name = null, SubNode = null });
+            var deserialized = serializer.Deserialize<ClassWithDefautString>(xml);
+
+            deserialized.Attribute.Should().BeNull();    //fail
+        }
+
+        [Fact]
+        public void ShouldPreserveNullObjectValueIfEmitWhenReturnsTrue()
+        {
+            var config = new ConfigurationContainer();
+            config.Emit(EmitBehaviors.Assigned); //no matter what we put here
+            config.ConfigureType<ClassWithDefautString>().Member(x => x.SubNode).EmitWhen(x => true);
+            var serializer = config.Create();
+
+            string xml = serializer.Serialize(new ClassWithDefautString() { Name = null, SubNode = null });
+            var deserialized = serializer.Deserialize<ClassWithDefautString>(xml);
+
+            deserialized.Attribute.Should().BeNull();    //fail
```

Comment:
Sanity check here... should be this be:
`deserialized.SubNode.Should().BeNull();`

?

As I mentioned I am not too good with Git, so what I've done is made a copy of this file locally and have been working on this from there.